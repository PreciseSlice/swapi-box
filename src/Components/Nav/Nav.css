.nav {
  display: flex;
  align-content: space-between;
  justify-content: center;
  margin: 30px; }
  .nav button {
    background-color: #4b074b;
    border: none;
    box-shadow: 0 0 5px 3px rgba(0, 59, 168, 0.8);
    font-family: 'Oswald', sans-serif;
    font-size: 1.6em;
    height: 60px;
    letter-spacing: 0.15em;
    margin: 20px;
    width: 14%;
    outline: none; }
    .nav button:hover {
      background-color: #feda4a; }
      .nav button:hover a {
        color: #4b074b; }
  .nav a {
    color: #feda4a;
    text-decoration: none; }
