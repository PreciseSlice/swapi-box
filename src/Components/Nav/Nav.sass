.nav 
  display: flex
  align-content: space-between
  justify-content: center
  margin: 30px 

  button 
    background-color: lightgray
    font-family: 'Oswald', sans-serif
    margin: 20px
    font-size: 1.2em
    height: 35px
    letter-spacing: 0.15em
    outline: none
    
    // &:active
    //   color: #feda4a


  a 
    padding: 5px
    text-decoration: none
    &:active
      color: #feda4a
  

