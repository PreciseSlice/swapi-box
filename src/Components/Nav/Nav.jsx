import React from 'react';
import './Nav.css';
import { NavLink } from 'react-router-dom';

let FavCount = 0;

const Nav = () => {
  return (
    <div className="nav">
      <button>
        <NavLink to="/">Opening Crawl</NavLink>
      </button>
      <button>
        <NavLink to="/people">People</NavLink>
      </button>
      <button>
        <NavLink to="/planets">Planets</NavLink>
      </button>
      <button>
        <NavLink to="/vehicles">Vehicles</NavLink>
      </button>
      <button>
        <NavLink to="/favorites">View Favorites {FavCount}</NavLink>
      </button>
    </div>
  );
};

export default Nav;
