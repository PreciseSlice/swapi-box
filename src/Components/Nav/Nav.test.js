/* eslint-disable */
import React from 'react';
import ReactDOM from 'react-dom';
import Nav from './Nav';
import { shallow } from 'enzyme';

describe('Nav', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Nav favCount={4}/>)
  })

  it('Exist and matches snapshot', () => {
    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });

});