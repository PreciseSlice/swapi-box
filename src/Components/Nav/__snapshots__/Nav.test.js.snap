// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Exist and matches snapshot 1`] = `
<div
  className="nav"
>
  <button>
    <NavLink
      activeClassName="active"
      ariaCurrent="true"
      to="/"
    >
      scroll
    </NavLink>
  </button>
  <button>
    <NavLink
      activeClassName="active"
      ariaCurrent="true"
      to="/people"
    >
      people
    </NavLink>
  </button>
  <button>
    <NavLink
      activeClassName="active"
      ariaCurrent="true"
      to="/planets"
    >
      planets
    </NavLink>
  </button>
  <button>
    <NavLink
      activeClassName="active"
      ariaCurrent="true"
      to="/vehicles"
    >
      vehicles
    </NavLink>
  </button>
  <button>
    <NavLink
      activeClassName="active"
      ariaCurrent="true"
      to="/favorites"
    >
      favorites 
      4
    </NavLink>
  </button>
</div>
`;
